docs/guide/renderer/implementing_serverside.md

```diff
@@ -1,9 +1,45 @@
 
-*Details comming soon...*
\ No newline at end of file
+To get a very good understanding what it takes to implement your own renderer it is worth going throught he sources of our `next_gen_ui_rhds_renderer` as it's a good example of everything that needs doing. In the next sections of this guide we'll be referencing various pieces of this package to illustrate the approach.
+
+## A step by step guide to create a renderer plugin
+Those instructions will assume you're implementing the plugin as part of our NextGenUI repository fork but it's also possible to implement it completely separate, you'll just have to align the steps with your project structure, build scripts etc.
+
+1. Clone [NextGenUI](https://github.com/RedHat-UX/next-gen-ui-agent) repository or your fork of it.
+2. Create a subdirectory in `./libs` directory that will contain source files for your renderer. By convention we tend to name our renderer packages with the pattenr of `next_gen_ui_******_renderer` with an example of `next_gen_ui_rhds_renderer`. This naming convention has no technical consequence, so it's up to you whether you follow it or not.
+3. Add the `__init__.py` file and `README.md` files in your newly created plugin directory.
+4. Create the main entrypoint script for your renderer. It'll extend classes provided in [./libs/next_gen_ui_agent/renderer/base_renderer.py](https://github.com/RedHat-UX/next-gen-ui-agent/blob/main/libs/next_gen_ui_agent/renderer/base_renderer.py). The code follows Factory and Strategy programming patterns to ease undersatnding and usage of the base implementation. You can also keep following [./libs/next_gen_ui_rhds_renderer/rhds_renderer.py](https://github.com/RedHat-UX/next-gen-ui-agent/blob/main/libs/next_gen_ui_rhds_renderer/rhds_renderer.py) for reference during implementing.
+   1. Extend `StrategyFactory` abstract class and implement all the required methods. The `get_render_strategy` one is the key handler of rendering. It's responsible for switching between different rendering strategies depending on the component type. On its output a class inheriting from `RenderStrategyBase` is expected.
+   2. Now for every component renderable by your system you have to provide a renderer strategy. They can be shared between components or you can have some iheritance approach as you can se in RHDS Renderer. The most important part is to extend `RenderStrategyBase` and then provide instances of those classes from the factory. Please refer to code comments in `RenderStrategyBase` to understand purpose of each of methods included in this class. If you just extend the class and provide no implementation, your renderer will already work and provide the default JSON responses.
```

Comment:
Maybe mention here that every component has own metadata and they are documented in `../../spec/component.md` .
And we should also mention Hand-Build Components here, that renderer is necessary for every HBC, and provide link to HBC documentation `../data_ui_blocks/hand_build_components.md`. Please note that support for HBC is not implemented in RHDS renderer yet, and that I just realized that HBC related code in `libs/next_gen_ui_agent/renderer` is not correct ;-) We have to clear this area a bit ;-)